src/vpn-api/app.ts

```diff
@@ -35,34 +39,121 @@ import worker from './worker';
 	);
-	if (VPN_INSTANCE_COUNT > 1) {
-		logger.info(`spawning ${VPN_INSTANCE_COUNT} workers`);
-		_.times(VPN_INSTANCE_COUNT, i => {
-			const instanceId = i + 1;
-			const restartWorker = (code?: number, signal?: string) => {
-				if (signal != null) {
-					logger.error(
-						`open-balena-vpn worker-${instanceId} killed with signal ${signal}`,
-					);
-				}
-				if (code != null) {
-					logger.error(
-						`open-balena-vpn worker-${instanceId} exited with code ${code}`,
-					);
-				}
-				cluster.fork({ VPN_INSTANCE_ID: instanceId }).on('exit', restartWorker);
-			};
-			restartWorker();
-		});
-	}
+	logger.info(`spawning ${VPN_INSTANCE_COUNT} workers`);
+	_.times(VPN_INSTANCE_COUNT, i => {
+		const instanceId = i + 1;
+		const restartWorker = (code?: number, signal?: string) => {
+			if (signal != null) {
+				logger.error(
+					`open-balena-vpn worker-${instanceId} killed with signal ${signal}`,
+				);
+			}
+			if (code != null) {
+				logger.error(
+					`open-balena-vpn worker-${instanceId} exited with code ${code}`,
+				);
+			}
+			cluster.fork({ VPN_INSTANCE_ID: instanceId }).on('exit', restartWorker);
+		};
+		restartWorker();
+	});
+
+	// track session stats from workers
+	const sessions: {
+		[key: string]: {
+			ts: number;
+			rxBytes: number;
+			txBytes: number;
+		};
+	} = {};
+
+	cluster.on('message', (_worker, msg) => {
+		const { uuid, ts, rxBytes, txBytes } = msg;
+		sessions[uuid] = { ts, rxBytes, txBytes };
+	});
+
+	// setup metrics for prometheus
+	metrics.describe(
+		'vpn_sessions_rx_bytes',
+		'histogram of rx bytes per vpn session',
+		{
+			// 1kb, 1mb, 10mb, 100mb, 500mb, 1gb, 10gb, 50gb, 100gb, 150gb, 200gb, 250gb, 500gb, 1tb
+			buckets: [
+				1024,
+				1024 * 1024,
+				10 * 1024 * 1024,
+				100 * 1024 * 1024,
+				500 * 1024 * 1024,
+				1024 * 1024 * 1024,
+				10 * 1024 * 1024 * 1024,
+				50 * 1024 * 1024 * 1024,
+				100 * 1024 * 1024 * 1024,
+				150 * 1024 * 1024 * 1024,
+				200 * 1024 * 1024 * 1024,
+				250 * 1024 * 1024 * 1024,
+				500 * 1024 * 1024 * 1024,
+				1024 * 1024 * 1024 * 1024,
+			],
+			labelNames: [],
+		},
+	);
+	metrics.describe(
+		'vpn_sessions_tx_bytes',
+		'histogram of tx bytes per vpn session',
+		{
+			buckets: [
+				1024,
+				1024 * 1024,
+				10 * 1024 * 1024,
+				100 * 1024 * 1024,
+				500 * 1024 * 1024,
+				1024 * 1024 * 1024,
+				10 * 1024 * 1024 * 1024,
+				50 * 1024 * 1024 * 1024,
+				100 * 1024 * 1024 * 1024,
+				150 * 1024 * 1024 * 1024,
+				200 * 1024 * 1024 * 1024,
+				250 * 1024 * 1024 * 1024,
+				500 * 1024 * 1024 * 1024,
+				1024 * 1024 * 1024 * 1024,
+			],
+			labelNames: [],
+		},
+	);
+
+	const app = express();
+	app.disable('x-powered-by');
+	app.get('/ping', (_req, res) => res.send('OK'));
+	app.use(morgan('combined'));
+	app.get('/metrics', (_req, res) => {
+		// res.writeHead(200, { 'Content-Type': 'text/plain' });
+		_.map(
+			_.filter(
+				sessions,
+				session =>
+					new Date().getTime() - session.ts <
```

Comment:
Can you grab the time just once rather than each loop (it's a surprisingly expensive operation).  You can also do it as:
```ts
const x = Date.now() - VPN_BYTECOUNT_INTERVAL * 1000 * 1.5
_.filter(sessions, session => session.ts > x)